enterprise/core-edge/src/main/java/org/neo4j/coreedge/raft/log/segmented/Terms.java

```diff
@@ -107,7 +107,30 @@ synchronized void truncate( long fromIndex )
     synchronized void prune( long upToIndex )
     {
         min = max( upToIndex, min );
-        // could also prune out array
+
+        int offset = find( min );
+
+        if ( offset == -1 )
+        {
+            return;
+        }
+
+        size = 1;
+        indexes = Arrays.copyOfRange( indexes, offset, indexes.length );
+        terms = Arrays.copyOfRange( terms, offset, terms.length);
+    }
+
+    private int find( long min )
+    {
+        for ( int i = 0; i < indexes.length; i++ )
+        {
+            if ( indexes[i] == min )
```

Comment:
The indexes define ranges, so you can't have an exact comparison. You need to find the lowest i with index[i] >= min and then prune out those that precede it, 0..(i - 1).